### PR TITLE
Integrate system tables with new processor

### PR DESCRIPTION
**Pull request by zhang2014**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Integrate system tables with new processor

## Changelog

- New Feature

## Related Issues

part #3931 

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by zhang2014** [comment]
after this pr. all system tables can work in the new processor framework.

**Review by sundy-li — APPROVED** [review]

File: query/src/pipelines/new/executor/pipeline_runtime_executor.rs [hunk 1]
```diff
@@ -24,48 +24,60 @@ use crate::pipelines::new::pipeline::NewPipeline;
...
+                        }
                     }
                 }
-            });
+            }));
         }
+
+        while let Some(join_handle) = threads.pop() {
+            // flatten error.
+            match join_handle.await {
```
Comment (sundy-li) [inline query/src/pipelines/new/executor/pipeline_runtime_executor.rs:71]: However, Rust futures won't do any work until they're actively .await !

**Comment by BohuTANG** [comment]
> after this pr. all system tables can work in the new processor framework.
Cool!
Seems many of the interpreter logic should be moved to a transform(or processor), then we can track and profile them under the new processor framework.
For example: insert(... select) and delete etc, then we can visualize the profile easily.

**Pull request merged**